src/app/models/AssociationDecorators.ts

```diff
@@ -11,28 +12,32 @@ import { User } from "./User";
  * Creates an association between the ownerId and its user model
  */
 export function owner<Parent extends AbstractModel & { ownerId?: Id }>() {
-  return hasOne<Parent, User>(ACCOUNT, "ownerId");
+  const key: keyof Parent = "ownerId";
+  return hasOne<Parent, User>(ACCOUNT, key as any);
```

Comment:
why is there a need to cast back to `any`?

Reply:
Yes, for some reason typescript cant see that the key in this case will be used to return an `Id`. Couldn't figure out how to get typescript to like it so i added some intermediate typing and then changed it to any to pass the typecheck.

Reply:
Ah that makes the rest of it make sense now. It'll do 👍